### PR TITLE
Music: fix empty timeline

### PR DESCRIPTION
**Pull request by breville**
This fixes a bug in which a timeline with no elements was almost infinitely scrollable horizontally.

### Before

Prior to this change, an empty timeline had a really big invisible element for the function bounds, allowing the user to scroll almost infinitely into empty space.

Screenshot 2024-04-23 at 1 05 21 PM

### After

With this change, the function bounds are no longer rendered if there's nothing inside them.

Screenshot 2024-04-23 at 1 05 44 PM

The function bounds still work appropriately.

Screenshot 2024-04-23 at 1 05 48 PM

### CODE REVIEW

File: apps/src/music/views/TimelineSimple2Events.tsx [hunk 1]
```diff
@@ -110,24 +119,34 @@ const TimelineSimple2Events: React.FunctionComponent<
...
-              top: 32 + uniqueFunction.top * eventHeight,
-              height:
-                (uniqueFunction.bottom - uniqueFunction.top) * eventHeight - 3,
-            }}
-          >
-            &nbsp;
-          </div>
-        ))}
+        {uniqueFunctionExtentsArray
+          .filter(uniqueFunction => uniqueFunction)
```
Comment (sanchitmalhotra126) [inline apps/src/music/views/TimelineSimple2Events.tsx]: minor - perform the `filter` in the `useMemo` above where `uniqueFunctionExtentsArray` is assigned, to save on the formatting change here?

**Review by sanchitmalhotra126 — COMMENTED** [review]
Nice catch!

File: apps/src/music/views/TimelineSimple2Events.tsx (hunk 1 above)
Comment (breville) [reply apps/src/music/views/TimelineSimple2Events.tsx]: Good question.  This prompted a bigger refactor, which is now committed.

**Review by sanchitmalhotra126 — APPROVED** [review]
Nice!

**Pull request merged**